### ISSUE TITLE
GraalVM native with kotlin issue

### ISSUE THREAD

**Issue by GrapeBaBa** [oracle/graal#8319]
**Describe the issue**
Our project https://github.com/optimism-java/hildr using https://github.com/libp2p/jvm-libp2p as dependency. When build as a jar, it works fine. After build as native, a method https://github.com/libp2p/jvm-libp2p/blob/24982eee203f72a96769fc103195c9807c61f425/libp2p/src/main/kotlin/io/libp2p/pubsub/PubsubApiImpl.kt#L64 seems not be called and no exception thrown.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. clone https://github.com/optimism-java/hildr
2. follow [here](https://github.com/optimism-java/hildr?tab=readme-ov-file#build-native)

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **[e.g. EE 19.3, CE 20.0, CE 20.1.0-dev-20200304_0848]**
 - JDK major version: graal-jdk-21.0.2
 - OS: macOS or linux
 - Architecture: amd64

**More details**
Add any other information about the problem here. Especially important are stack traces or log output. Feel free to link to gists or to screenshots if necessary.
```
```

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reaching out, did you try generating the configuration files using our Tracing agent? If not, please do so and check this doc for details: https://www.graalvm.org/22.3/reference-manual/native-image/metadata/AutomaticMetadataCollection/

**Comment by vjovanov** [comment]
Thanks for reporting this issue, adding more instructions would help reproduce this issue: which subproject, how to run, etc. Also, does this happen with the statically linked binary or with the regular one? 

I would run the build with the latest GraalVM version (snapshot ideally) and add the flag `-H:ThrowMissingRegistrationErrors=` to see if it fails. Also, it would be good to have the output of `-H:+PrintClassInitialization`.

**Comment by GrapeBaBa** [comment]
> Thanks for reporting this issue, adding more instructions would help reproduce this issue: which subproject, how to run, etc. Also, does this happen with the statically linked binary or with the regular one?
> 
> I would run the build with the latest GraalVM version (snapshot ideally) and add the flag `-H:ThrowMissingRegistrationErrors=` to see if it fails. Also, it would be good to have the output of `-H:+PrintClassInitialization`.

will do and feedback.

**Comment by GrapeBaBa** [comment]
> Hi, Thank you for reaching out, did you try generating the configuration files using our Tracing agent? If not, please do so and check this doc for details: https://www.graalvm.org/22.3/reference-manual/native-image/metadata/AutomaticMetadataCollection/

Yes, we generated the files and works fine. Only this code seems not execution when compile to native.

**Comment by GrapeBaBa** [comment]
> Thanks for reporting this issue, adding more instructions would help reproduce this issue: which subproject, how to run, etc. Also, does this happen with the statically linked binary or with the regular one?
> 
> I would run the build with the latest GraalVM version (snapshot ideally) and add the flag `-H:ThrowMissingRegistrationErrors=` to see if it fails. Also, it would be good to have the output of `-H:+PrintClassInitialization`.

@vjovanov it is easy to build and run, however it need some env config which include API key, how can I share it to you.

**Comment by vjovanov** [comment]
`-H:ThrowMissingRegistrationErrors=` is a build-time flag that throws when there is missing reflection metadata. It can be used in conjunction with `-H:MissingRegistrationReportingMode=Warn` to print missing elements, or `-H:MissingRegistrationReportingMode=Exit` to hard fail on missing elements.

**Comment by GrapeBaBa** [comment]
I feel it is possible a problem with network layer since it is a p2p library based UDP. Is there anything I need to pay attention to in such case?

**Comment by vjovanov** [comment]
Is the reflection configured properly? No exceptions when you use: `-H:ThrowMissingRegistrationErrors=`? 


If no exceptions, you can send me the keys over the community slack: [Slack channels](https://www.graalvm.org/slack-invitation/)
